django/contrib/postgres/operations.py

```diff
@@ -164,3 +164,65 @@ def database_backwards(self, app_label, schema_editor, from_state, to_state):
             to_model_state = to_state.models[app_label, self.model_name_lower]
             index = to_model_state.get_index_by_name(self.name)
             schema_editor.add_index(model, index, concurrently=True)
+
+
+class CreateCollation(Operation):
+    reversible = True
+
+    def __init__(self, name, *, locale=None, provider=None, deterministic=True):
+        self.name = name
+        self.locale = locale
+        self.provider = provider
+        self.deterministic = deterministic
+
+    def state_forwards(self, app_label, state):
+        pass
+
+    def database_forwards(self, app_label, schema_editor, from_state, to_state):
+        if (
+            schema_editor.connection.vendor != 'postgresql' or
+            not router.allow_migrate(schema_editor.connection.alias, app_label)
+        ):
+            return
+        if (
+            self.deterministic is False and
+            not schema_editor.connection.features.supports_non_deterministic_collations
+        ):
+            raise NotSupportedError('Non-deterministic collations require PostgreSQL 12 or above')
```

Comment:
```suggestion
            raise NotSupportedError('Non-deterministic collations require PostgreSQL 12+.')
```

Also, this exception isn't tested.